### PR TITLE
Add `RewardingBuilder`, `CertifyingBuilder`, and their friends

### PR DESCRIPTION
**Pull request by SeungheonOh**
Adding these as we see some need for StakingValidator samples.

Checkers can be added later I think. If not I'll throw them in.

### CODE REVIEW

**Comment by peter-mlabs** [comment]
Yep, checkers can be added later -- its not quite worth doing anything besides the bare minimum right now, becuase PCB will probably go through a broader rewrite. I'll open a backlog ticket for these.

File: src/Plutarch/Context/Certifying.hs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+-- | @since 2.8.0
+instance Monoid CertifyingBuilder where
+  mempty = CB mempty Nothing
+
+-- | @since 2.8.0
+instance Builder CertifyingBuilder where
+  _bb = #inner
+  pack x = set #inner x (mempty :: CertifyingBuilder)
+
+instance Normalizer CertifyingBuilder where
```
Comment (peter-mlabs) [inline src/Plutarch/Context/Certifying.hs:100]: adding a `since` here for you

**Review by peter-mlabs — APPROVED** [review]

**Comment by SeungheonOh** [comment]
By broad rewrite, do you mean monad based one? For modification support?

**Comment by peter-mlabs** [comment]
Yeah, the "templating" rewrite.

**Pull request merged**